### PR TITLE
sleef: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**
TODO

- [ ] v1: `Package with 'shared=False' option did not contain any static artifact`

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

File: recipes/sleef/all/conanfile.py [hunk 1]
```diff
@@ -24,75 +28,79 @@ class SleefConan(ConanFile):
...
-        self._cmake.definitions["ENABLE_ALTDIV"] = False
-        self._cmake.definitions["ENABLE_ALTSQRT"] = False
-        self._cmake.definitions["DISABLE_FFTW"] = True
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
```
Comment (uilianries) [inline recipes/sleef/all/conanfile.py:66]: Some options changed in 3.6, including for testing. Please, check in the new release.

**Review by uilianries — COMMENTED** [review]
Thank you for your PR, please, take a look in my review.

File: recipes/sleef/all/conanfile.py [hunk 2]
```diff
@@ -24,75 +30,92 @@ class SleefConan(ConanFile):
...
+            if cross_building(self):
+                # Fails with "No rule to make target `/bin/mkrename'"
+                # https://github.com/shibatch/sleef/issues/308
+                raise ConanInvalidConfiguration(f"{self.ref} does not support cross-building with apple-clang")
+            if Version(self.version) < "3.6" and self.settings.arch == "armv8":
+                # clang: error: the clang compiler does not support '-march=armv7-a'
+                # clang: warning: argument unused during compilation: '-mfpu=vfpv4' [-Wunused-command-line-argument]
+                # clang: warning: argument unused during compilation: '-arch arm64' [-Wunused-command-line-argument]
+                # clang: warning: argument unused during compilation: '-mmacosx-version-min=11.0' [-Wunused-command-line-argument]
+                raise ConanInvalidConfiguration(f"armv8 is not supported for {self.ref}")
```
Comment (uilianries) [inline recipes/sleef/all/conanfile.py:61]:
```suggestion
                raise ConanInvalidConfiguration(f"{self.ref} does not support Mac M1. Please, use {self.name} version >=3.6.")
```

Only a small detail, so people can know about which version supports M1 without reading the recipe.

**Review by uilianries — APPROVED** [review]
LGTM. Thank you for updating to 3.6

**Review by uilianries — APPROVED** [review]

**Pull request merged**